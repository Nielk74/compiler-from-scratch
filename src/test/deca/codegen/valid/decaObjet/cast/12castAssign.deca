// Description:
//    Test de cast pour l'expression assign
//
// Resultats:
//      12castAssign.expected
//
// Historique:
//    cree le 21/06/2022

class A {
    int x=1;
    void m(){
        println("from A");
    }
}
class B extends A{
    int x=2;
    void m(){
        println("from B");
    }
}
class C extends B{
    int x=3;
    void m(){
        println("from C");
    }
}


{
    A a;
    B b;
    B b1 = (B)(a = b = new C());

    a.m();
    b.m();
    b1.m();
    println(a.x);
    println(b.x);
    println(b1.x);
}
