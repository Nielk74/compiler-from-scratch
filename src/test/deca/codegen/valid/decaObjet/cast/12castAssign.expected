from C
from C
from C
1
2
2
